src/compiler/utils/full_char_code_at.rs

```diff
@@ -0,0 +1,11 @@
+//Named full_char_code_at in javascript compiler
+pub fn word(str: &str, i: usize) -> u8 {
+    let code = &str[i..i+1].as_bytes()[0];
+    
+    if *code <= 0xd7ff || *code >= 0xe000 {
+        return *code
+    }
+
+    let next = &str[i + 1..i+2].as_bytes()[0];
+    (code << 10) + next - 0x35fdc00
+}
```

Comment:
Isn't the `str` UTF-8 encoded? If I'm correct, `&str[i..i+1]` isn't the same as `str.charCodeAt(i)`.

If this function is for finding the Unicode code point, maybe this is suffice:

```suggestion
pub fn full_char_code_at(str: &str, i: usize) -> u32 {
    str.chars().nth(i).unwrap() as u32
}
```

But I guess, the original code may rely on a lot JS string indexings so we'd better see the usage on the call site for better implementation. Creating an iterator every time we do this doesn't feel so good :)

The renaming to `word` doesn't make sense IMO.

Reply:
Thank you for this suggestion this is a much better implementation

Reply:
You are right `str` is UTF-8 encoded made the appropriate changes.